### PR TITLE
[JV15-258] paginate transactions logs fetching

### PR DESCRIPTION
**Pull request by davodesign84**

### CODE REVIEW

File: libraries/ts/src/margin/marginClient.ts [hunk 1]
```diff
@@ -237,11 +237,19 @@ export class MarginClient {
...
-    const transactions = await provider.connection.getParsedTransactions(
-      signatures.map(s => s.signature),
-      "confirmed"
-    )
-    const jetTransactions = MarginClient.filterTransactions(transactions, config)
+    const jetTransactions: ParsedTransactionWithMeta[] = []
+    let page = 0
+    // fetch up to 100 JET transactions not the cleanest solution but should be an ok patch until we have a data api
+    while (jetTransactions.length < 100) {
+      const paginatedSignatures = signatures.slice(page * 100, (page + 1) * 100)
```
Comment (jetjake) [inline libraries/ts/src/margin/marginClient.ts:244]: Can we move these magic numbers to Margin Client initialization or method parameters with default of 100?

**Review by jetjake — COMMENTED** [review]
One comment, also a test is failing.

File: libraries/ts/src/margin/marginClient.ts (hunk 1 above)
Comment (davodesign84) [reply libraries/ts/src/margin/marginClient.ts:244]: Yeah working on making the tests pass for now :) Adding params as part of this.

**Review by jetjake — APPROVED** [review]

File: libraries/ts/src/margin/marginClient.ts (hunk 1 above)
Comment (davodesign84) [reply libraries/ts/src/margin/marginClient.ts]: Addressed now, there was also a stupid problem with the logic 😬  Fixed both

**Pull request merged**